### PR TITLE
[MM-46568] - AD-LDAP upsell banner

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR adds an ADLDAP upsell banner for professional self-managed workspaces when they visit the user-groups list modal

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46571
https://mattermost.atlassian.net/browse/MM-46569
https://mattermost.atlassian.net/browse/MM-46568

#### Screenshots
Screenshot 2022-09-28 at 17 06 57
Screenshot 2022-09-28 at 17 07 11

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/user_groups_modal/user_groups_modal.scss [hunk 1]
```diff
@@ -11,6 +11,118 @@
...
+                font-weight: 400;
+                line-height: 20px;
+            }
+
+            .btns-container {
+                display: flex;
+                justify-content: right;
+                gap: 10px;
+
+                .confrim-btn {
```
Comment (Afsoon) [inline components/user_groups_modal/user_groups_modal.scss:101]: typo
```suggestion
                .confirm-btn {
```

File: components/user_groups_modal/ad_ldap_upsell_banner.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    const openSelfHostedLink = () => {
+        window.open(LicenseLinks.CONTACT_SALES, '_blank');
+    };
+
+    const confirmBanner = (
+        <div className='ad_ldap_upsell_confirm'>
+            <div className='upsell-confirm-backdrop'/>
+            <div className='upsell-confirm-foreground'>
+                <p className='title'>{formatMessage({id: 'adldap_upsell_banner.confirm.title', defaultMessage: 'Your trial has started!'})}</p>
+                <p className='subtitle'>{formatMessage({id: 'adldap_upsell_banner.confirm.license_trial', defaultMessage: 'Welcome to your Mattermost Enterprise trial! It expires on {endDate}. You now have access to high-security Enterprise features, for free.'}, {endDate: moment(currentLicenseEndDate).tz(getBrowserTimezone()).format('MMMM Do YYYY')})}</p>
```
Comment (Afsoon) [inline components/user_groups_modal/ad_ldap_upsell_banner.tsx:55]: ❓ Why not use `intl.formatDate` for the date format instead hardcoded a structure?

**Review by Afsoon — APPROVED** [review]
LGTM just a few non blocked comments.

**Review by nickmisasi — APPROVED** [review]
LGTM!

**Comment by AGMETEOR** [comment]
Hello @yasserfaraazkhan . Gentle ping

**Comment by yasserfaraazkhan** [comment]
@AGMETEOR 
thanks for the changes. I couldn't add e2e for it, but the tests component tests covers most scenarios.

[MM-T5189](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/project/10302/testCase?folderId=1285126)

**Review by yasserfaraazkhan — APPROVED** [review]
🎉

**Comment by AGMETEOR** [comment]
> MM-T5189

Yes, there's only one scenario they don't cover. The journey from  when "Try free for 30 days" is clicked and the confirm modal opens

**Review by annelieseklein — APPROVED** [review]

**Comment by annelieseklein** [comment]
@AGMETEOR the only thing is the X icon should not be bold

**Review by yasserfaraazkhan — APPROVED** [review]
Hi Allan
I've tested the cloud scenarios locally and the flow is as expected

**Pull request merged**